lib/bolt/transport/winrm/connection.rb

```diff
@@ -183,6 +232,51 @@ def write_executable_from_content(dir, content, filename)
           write_remote_file(content, remote_path)
           remote_path
         end
+
+        private
+
+        # NTSTATUS https://msdn.microsoft.com/en-us/library/cc704588.aspx
+        STATUS_SUCCESS = 0
+        STATUS_LOGON_FAILURE = 0xC000006D
+
+        SMB_PORT = 445
+
+        def smb_client_login
+          return @client if @client
+
+          dispatcher = RubySMB::Dispatcher::Socket.new(smb_socket_connect)
+          @client = RubySMB::Client.new(dispatcher, smb1: true, smb2: true, username: @user, password: target.password)
+          status = @client.login
```

Comment:
Verify signing has been required by the server (by default), but allow downgrade (maybe).